onnxscript/function_libs/torch_lib/graph_building.py

```diff
@@ -725,10 +762,14 @@ def to_model_proto(
         )
 
         try:
-            onnx_model = onnx.shape_inference.infer_shapes(
-                onnx_model, check_type=True, strict_mode=False, data_prop=True
-            )
-            onnx.checker.check_model(onnx_model, full_check=True)
+            if not cache_model_to_disk:
+                # Only check the model if it is in memory.
+                # Otherwise the checker and shape_inference will fail because
```

Comment:
For shape inference, can we still load shape and element type from model file (not initializer files) and then run infer_shape?

Reply:
We could, but we also don’t need to because PyTorch supplies all the shape info.

Reply:
A drawback due to https://github.com/onnx/onnx/issues/5487, we don't have much inner node shape info left now that modules are functions.